backend/src/main/java/com/cruru/advice/GlobalExceptionHandler.java

```diff
@@ -1,76 +1,89 @@
 
-    private final UncatchedExceptionHandler handler;
+    private static final int STACK_TRACE_LIMIT = 5;
 
     @ExceptionHandler(CruruCustomException.class)
-    public ResponseEntity<ProblemDetail> handleCustomException(CruruCustomException e) {
-        HttpServletRequest request = getCurrentHttpRequest();
-        ExceptionLogger.info(request, e);
-
+    public ResponseEntity<Object> handleCustomException(CruruCustomException e) {
         ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(e.getStatus(), e.getMessage());
-        return ResponseEntity.of(problemDetail).build();
+
+        return handleExceptionWithMdc(e, problemDetail,
+                ex -> log.info("CRURU-EXCEPTION [errorMessage = {}]", ex.getMessage())
+        );
     }
 
     @ExceptionHandler(Exception.class)
-    public ResponseEntity<ProblemDetail> handleUnexpectedException(Exception e, WebRequest request) {
-        ProblemDetail problemDetail = handleException(e, request);
-        HttpStatus statusCode = HttpStatus.valueOf(problemDetail.getStatus());
-        if (statusCode.is5xxServerError()) {
-            ExceptionLogger.error(problemDetail);
-        } else {
-            ExceptionLogger.warn(problemDetail);
-        }
-        ProblemDetail detailsToSend = ProblemDetail.forStatus(statusCode);
-        return ResponseEntity.of(detailsToSend).build();
+    public ResponseEntity<Object> handleUnexpectedException(Exception e) {
+        ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(HttpStatus.INTERNAL_SERVER_ERROR, "서버 오류입니다.");
+
+        return handleExceptionWithMdc(e, problemDetail,
+                ex -> log.error("SERVER-EXCEPTION [errorMessage = {}, stackTrace = {}]", ex.getMessage(),
+                        getLimitedStackTrace(ex))
+        );
     }
 
-    private ProblemDetail handleException(Exception e, WebRequest request) {
-        try {
-            ProblemDetail problemDetail = (ProblemDetail) Objects.requireNonNull(handler.handleException(e, request))
-                    .getBody();
-            problemDetail.setProperties(setDetails(getCurrentHttpRequest(), e, HttpStatus.INTERNAL_SERVER_ERROR));
-            return problemDetail;
-        } catch (Exception ex) {
-            return ProblemDetail.forStatusAndDetail(
-                    HttpStatus.INTERNAL_SERVER_ERROR,
-                    "예기치 못한 오류가 발생하였습니다."
-            );
+    private String getLimitedStackTrace(Exception e) {
+        StackTraceElement[] stackTraceElements = e.getStackTrace();
+        StringBuilder sb = new StringBuilder();
+        int length = Math.min(stackTraceElements.length, STACK_TRACE_LIMIT);
+
+        for (int i = 0; i < length; i++) {
+            sb.append(stackTraceElements[i].toString());
         }
+
+        if (stackTraceElements.length > STACK_TRACE_LIMIT) {
+            sb.append("... (stack trace truncated)");
+        }
+
+        return sb.toString();
     }
```

Comment:
stringBuilder 없이 람다와 joining을 통해 구현 가능하네요!

```suggestion
private String getLimitedStackTrace(Exception e) {
    StackTraceElement[] stackTraceElements = e.getStackTrace();
    int length = Math.min(stackTraceElements.length, STACK_TRACE_LIMIT);

    String stackTrace = IntStream.range(0, length)
            .mapToObj(i -> stackTraceElements[i].toString())
            .collect(Collectors.joining());

    if (stackTraceElements.length > STACK_TRACE_LIMIT) {
        stackTrace += "... (stack trace truncated)";
    }

    return stackTrace;
}
```

Reply:
명오방식 반영했습니다~